pkg/tagger/telemetry/telemetry.go

```diff
@@ -67,3 +72,21 @@ var (
 		[]string{}, "Number of of times the tagger has received a notification with a group of events",
 		telemetry.Options{NoDoubleUnderscoreSep: true})
 )
+
+var queriesForCardinality []telemetry.SimpleCounter
+
+func init() {
+	queriesForCardinality = make([]telemetry.SimpleCounter, collectors.HighCardinality+1)
+	queriesForCardinality[collectors.LowCardinality] = queries.WithValues(collectors.LowCardinalityString)
+	queriesForCardinality[collectors.OrchestratorCardinality] = queries.WithValues(collectors.OrchestratorCardinalityString)
+	queriesForCardinality[collectors.HighCardinality] = queries.WithValues(collectors.HighCardinalityString)
+}
+
+// IncQueriesForCardinality increments the telemetry counter for number of tags queries with specific cardinality.
+func IncQueriesForCardinality(card collectors.TagCardinality) {
+	if card <= collectors.HighCardinality && queriesForCardinality[card] != nil {
+		queriesForCardinality[card].Inc()
+	} else {
+		queries.Inc(collectors.TagCardinalityToString(card))
```

Comment:
Can `card <= collectors.HighCardinality && queriesForCardinality[card] != nil` be false?

Reply:
Yes, in case we introduce new cardinality level but do not update the initialization code above.